### PR TITLE
Hosting onboarding: Customise home experience for hosting devs

### PR DESCRIPTION
**Pull request by vykes-mac**
This PR provides basic boilerplate to tailor the home page view experience for developers. The views will change depending on what will be agreed suitable for a developer to see on the home page.

Related to https://github.com/Automattic/dotcom-forge/issues/2629

## Proposed Changes

* Create custom quick links component tailored to developers
* Remove daily blogging prompts (done in the backend).

## Testing Instructions

- Apply patch D112799-code 
- Verify daily blog post prompt is now shown
- Verify developer quick links are shown

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/customer-home/cards/actions/quick-links-for-devs/index.jsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+					href={ `/plugins/${ siteSlug }` }
+					hideLinkIndicator
+					label={ translate( 'Explore Plugins' ) }
+					gridicon="plugins"
+				/>
+			) }
+			<ActionBox
+				href={ `/themes/${ siteSlug }` }
+				hideLinkIndicator
+				label={ translate( 'Explores themes' ) }
```
Comment (zaguiini) [inline client/my-sites/customer-home/cards/actions/quick-links-for-devs/index.jsx]:
```suggestion
				label={ translate( 'Explore themes' ) }
```

**Review by zaguiini — COMMENTED** [review]
Looking good 😎

File: client/my-sites/customer-home/cards/actions/quick-links-for-devs/index.jsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		,
+		flushDebouncedUpdateHomeQuickLinksToggleStatus,
+	] = useDebouncedCallback( updateHomeQuickLinksToggleStatus, 1000 );
+
+	const quickLinks = (
+		<div className="quick-links-for-devs__boxes quick-links__boxes">
+			<ActionBox
+				href={ `/hosting-config/${ siteSlug }#sftp-credentials` }
+				hideLinkIndicator
+				label={ translate( 'Setup SSH' ) }
```
Comment (mpkelly) [inline client/my-sites/customer-home/cards/actions/quick-links-for-devs/index.jsx]: Grammar nit: Should be `Set up SSH`.

**Review by mpkelly — COMMENTED** [review]
This is working for me on both new and existing sites. 

image

I noticed a small issue with the domain preview for RTL languages but that's a different PR. 
Screenshot 2566-06-06 at 14 30 02

**Comment by p-jackson** [comment]
Heads up that another way to test the view is to add `?view=VIEW_FOR_DEVS` to the URL in Calypso. This forces the API to return the specified view regardless of eligibility. Not a big deal for us, but very useful when it takes some time to set up the site conditions to return a particular view.

File: client/my-sites/customer-home/cards/actions/quick-links-for-devs/index.jsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+				hideLinkIndicator
+				label={ translate( 'Set up SSH' ) }
+				materialIcon="key"
+			/>
+			<ActionBox
+				href={ `/hosting-config/${ siteSlug }#staging-site` }
+				hideLinkIndicator
+				label={ translate( 'Create staging site' ) }
+				gridicon="science"
+			/>
```
Comment (p-jackson) [inline client/my-sites/customer-home/cards/actions/quick-links-for-devs/index.jsx:51]: Should these actions also only be rendered by the `isAtomic` prop, like the Copy site action is?

Alternatively we could change the eligibility logic on the backend to only return this view when the site is atomic.

**Review by p-jackson — COMMENTED** [review]
Just dropping by to say the PR looks good.

We could take the opportunity of having a new component to switch from using `connect()` to the `useSelector()` and `useDispatch()` hooks.

File: client/my-sites/customer-home/cards/actions/quick-links-for-devs/index.jsx (hunk 3 above)
Comment (vykes-mac) [reply client/my-sites/customer-home/cards/actions/quick-links-for-devs/index.jsx:51]: I was thinking about that then I thought maybe show them still when dev creates a free site and let them link to the hosting page to show the upsell banner.

Comment (p-jackson) [reply client/my-sites/customer-home/cards/actions/quick-links-for-devs/index.jsx:51]: That seems fine too. It doesn't really feel like a "quick link" if it takes users to an upsell, but it's consistent with all the other links where we send users to hosting config.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8046240

Thank you @vykes-mac for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8046240).

**Pull request merged**